hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/AbstractSessionCredentialsProvider.java

```diff
@@ -132,13 +134,15 @@ public AWSCredentials getCredentials() throws SdkBaseException {
     }
     if (awsCredentials == null) {
       throw new CredentialInitializationException(
-          "Provider " + this + " has no credentials");
+          "Provider " + this + " has no credentials: " +
+             (initializationException!=null ? initializationException.getMessage() : ""),
```

Comment:
make it initializationException.toString(), so things like NullPointerException get mentions. also, put spaces around the !=